### ISSUE TITLE
Triple click doesn't select paragraph; CaretSelectionBindImpl.moveToParEnd wrongly calculates position

### ISSUE THREAD

**Issue by alesp** [FXMisc/RichTextFX#742]
Triple clicking a paragraph should select it. This works fine if you triple click the first paragraph. If you click any other paragraph, it will select some text before that paragraph.

This could be due to the fact that CaretSelectionBindImpl.moveToParEnd doesn't take into account that the cursor may not be in the first paragraph. It calculates new position as:

int newPos = getArea().getParagraphLength(getParagraphIndex());

This position is ok _relative to_ the current paragraph, but the next line (moveTo) apparently expects _absolute position_:

moveTo(newPos, selectionPolicy);

Compare that with moveTo in moveToParStart:

moveTo(getPosition() - getColumnPosition(), selectionPolicy);

In other words, moveTo in moveToParEnd should probably be:

moveTo(newPos + getPosition() - getColumnPosition(), selectionPolicy);

**Comment by JordanMartinez** [comment]
Ha! Wow... Yeah, that's a bug!

**Comment by JordanMartinez** [comment]
Do you want to fix it via a PR? Your proposed solution sounds good to me.

**Comment by JordanMartinez** [comment]
By the way, this bug can be fixed by overriding the default behavior
````java
Nodes.addInputMap(area, InputMap.consume(
    mouseClick().onlyIf(e -> e.getButton == PRIMARY && e.getClickCount == 3,
    e -> {
        CaretSelectionBind c = area.getCaretSelectionBind();
        int paragraphLength = area.getParagraphLength(c.getParagraphIndex());
        SelectionPolicy selPolicy = SelectionPolicy.ADJUST // might be Extend for mac users
        area.moveTo(c.getPosition() - c.getColumnPosition() + parLength, selPolicy);
})
````

**Comment by alesp** [comment]
I won't do a PR because I don't know the code well enough to know it won't affect something else.

Thank you for the workaround!

**Comment by JordanMartinez** [comment]
> I won't do a PR because I don't know the code well enough to know it won't affect something else.

We've got pretty extensive testing, so I wouldn't be afraid of that. Besides, you'd likely just be replacing the old code with the code I have above. The only other change may be to add another test to insure the code works on paragraphs that aren't the first one in the area.

**Comment by JordanMartinez** [comment]
Latest snapshot has the bug fix
